### ISSUE TITLE
[naming-convention] camelCase format doesn't work if prefix is configured

### ISSUE THREAD

**Issue by villelahdenvuo** [typescript-eslint/typescript-eslint#2187]
**Repro**

```JSON
{
  "rules": {
	"@typescript-eslint/naming-convention": [
		"error",
		{
			"selector": "variable",
			"types": ["boolean"],
			"format": ["camelCase"],
			"prefix": ["is", "should", "has", "can", "did", "will"]
		}
	]
  }
}
```

```TS
const isStore = isPlainObject(store) && isPlainObject(store.storefront) && isPlainObject(store.locale);
```

**Expected Result**

No error.

**Actual Result**

Variable name isStore must match one of the following formats: camelCase eslint@typescript-eslint/naming-convention

**Additional Info**

![image](https://user-images.githubusercontent.com/633524/84020480-28e30d00-a983-11ea-8456-94da100f0339.png)

If I remove the prefix option the format rule works.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `13.13.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
This is intentional. See: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#format-options

If you define a prefix, it is removed from the name before checking the format. This is done because otherwise you cannot do something like `StrictPascalCase` with a prefix (i.e. `TMyType` would never be able to match `StrictPascalCase`, due to the two capitals).

This is also shown in the example for configuring boolean names are prefixed with a verb:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-that-boolean-variables-are-prefixed-with-an-allowed-verb

**Comment by villelahdenvuo** [comment]
@bradzacher Thanks, that definitely makes sense. To avoid this issue in the future I made a PR to note this more visibly in the docs. ✨
